vpd-tool/include/tool_utils.hpp

```diff
@@ -125,5 +126,77 @@ inline std::string getPrintableValue(const types::BinaryVector& i_keywordValue)
 
     return l_oss.str();
 }
+
+/**
+ * @brief API to save keyword's value on file.
+ *
+ * API writes keyword's value on the given file path. If the data is in hex
+ * format, API strips '0x' and saves the value on the given file.
+ *
+ * @param[in] i_filePath - File path.
+ * @param[in] i_keywordValue - Keyword's value.
+ *
+ * @return - true on successfully writing to file, false otherwise.
+ */
+inline bool saveToFile(const std::string& i_filePath,
+                       const std::string& i_keywordValue)
+{
+    try
+    {
+        std::string l_keywordValue{i_keywordValue};
+
+        std::ofstream l_outPutFileStream(i_filePath, std::ofstream::binary);
+        l_outPutFileStream.exceptions(std::ifstream::badbit |
+                                      std::ifstream::failbit);
+
+        if (l_outPutFileStream.is_open())
+        {
+            if (i_keywordValue.substr(0, 2).compare("0x") ==
+                constants::STR_CMP_SUCCESS)
+            {
+                l_keywordValue = i_keywordValue.substr(2);
+            }
+
+            l_outPutFileStream.write(l_keywordValue.c_str(),
+                                     l_keywordValue.size());
+        }
+        else
+        {
+            // TODO: Enable logging when verbose is enabled.
+            // std::cerr << "Error occured while opening output file: " <<
+            // i_filePath << std::endl;
+            return false;
+        }
+
+        l_outPutFileStream.close();
+        return true;
+    }
+    catch (const std::exception& l_ex)
+    {
+        // TODO: Enable logging when verbose is enabled.
```

Comment:
if file is open  and we hit an exception in write , don't we need to close file  here also ?